website/static/js/apiApplication.js

```diff
@@ -0,0 +1,433 @@
+            });
+        }.bind(this));
+        return request;
+    },
+    createApplication: function () {
+        var request = this.client.createOne(this.appData());
+        request.done(function (dataObj) {
+            this.appData(dataObj);
+            this.originalValues(dataObj.serialize());
+
+            this.changeMessage(language.apiOauth2Application.creationSuccess, 'text-success', 5000);
+            this.apiDetailUrl(dataObj.apiDetailUrl); // Toggle ViewModel --> act like a display view now.
+            historyjs.replaceState({}, '', dataObj.webDetailUrl);  // Update address bar to show new detail page
+        }.bind(this));
+
+        request.fail(function (xhr, status, error) {
+            $osf.growl('Error',
+                       language.apiOauth2Application.dataSendError,
+                       'danger');
+
+            Raven.captureMessage('Error registering new OAuth2 application', {
+                url: this.apiDetailUrl,
+                status: status,
+                error: error
+            });
+        }.bind(this));
+    },
+    submit: function () {
+        // Validate and dispatch form to correct handler based on view type
+        if (!this.appData().isValid()) {
+            // Turn on display of validation messages
+            this.showMessages(true);
+        } else {
+            this.showMessages(false);
+            if (this.isCreateView()) {
+                this.createApplication();
+            } else {
+                this.updateApplication();
+            }
+        }
+    },
+    deleteApplication: function () {
+        var appData = this.appData();
+        bootbox.confirm({
+            title: 'Deactivate application?',
+            message: language.apiOauth2Application.deactivateConfirm,
+            callback: function (confirmed) {
+                if (confirmed) {
+                    var request = this.client.deleteOne(appData );
+                    request.done(function () {
+                        this.allowExit(true);
+                        // Don't let user go back to a deleted application page
+                        historyjs.replaceState({}, '', this.webListUrl);
+                        this.visitList();
+                    }.bind(this));
+                    request.fail(function () {
+                            $osf.growl('Error',
+                                       language.apiOauth2Application.deactivateError,
+                                       'danger');
+                    }.bind(this));
+                }
+            }.bind(this),
+            buttons:{
+                confirm:{
+                    label:'Deactivate',
+                    className:'btn-danger'
+                }
+            }
+        });
+    },
+    visitList: function () {
+        window.location = this.webListUrl;
+    },
+    cancelChange: function () {
+        if (!this.dirty()) {
+            this.visitList();
+        } else {
+            bootbox.confirm({
```

Comment:
This modal was added in direct response to UI feedback by @brianjgeiger here: https://github.com/CenterForOpenScience/osf.io/pull/3082#issuecomment-125242396

Additionally, the style guide calls for modal confirmation when there is unsaved content or settings:
https://centerforopenscience.github.io/osf-style/ui.html#confirmations

I am unsure how to handle this suggestion.